src/linking/static_lifetime_init.cpp

```diff
@@ -157,3 +160,21 @@ void static_lifetime_init(
     }
   }
 }
+
+void recreate_initialize_function(
+  goto_modelt &goto_model,
+  message_handlert &message_handler)
+{
+  if(goto_model.unload(INITIALIZE_FUNCTION) != 0)
```

Comment:
Would it be sensible to make it a precondition that the initialize function exists, as opposed to doing a silent no-op?

Reply:
I have now changed the usage model according to your suggestion.